### DISCUSSION TITLE
Can I Change entire theme at a runtime, not just ThemeVariant?

### DISCUSSION THREAD

**Discussion by shin0kaze** [AvaloniaUI/Avalonia#17334] [Q&A] [answered]
Can I Replace entire theme, like changing between Material, Fluent and Simple themes?
I Tried to clear Styles of Application, then add styles of new theme, but then on click I got this Exception: 
`The control ItemsPresenter (Name = PART_ItemsPresenter) already has a visual parent ScrollContentPresenter (Name = PART_ContentPresenter) while trying to add it as a child of ScrollContentPresenter (Name = PART_ContentPresenter, Host = ScrollViewer).`.

I think I need to clear visual tree, am I right? How can I do that?

**Comment by stevemonaco** [comment, marked as answer]
IIRC, the ControlCatalog closes and recreates the Window when changing themes from Simple <-> Fluent. Probably applies the theme from code-behind, too. In practice of a well-designed app, changing themes is not as viable as one might think. Each theme is allowed to be opinionated about its control sizes, margins, and padding. eg. Fluent is way more spacious (even on Compact density) than Simple because it was built to be touch-friendly.

You must be careful applying any explicit positioning tweaks in your app views in one theme...because it probably won't look the same on all of them. You could create `ResourceDictionary`s that you swap that contains app view spacings for each theme per-dictionary, but you're still wrestling with inconsistent control sizes and inconsistent internal spacing.

**Reply by shin0kaze** [reply]
Thanks for the Reply! For some reason I cannot build Avalonia Examlpes.
It will be easier to just load the choosen theme on next startup in main window constructor, then load a custom styles to Unify UI controls positions look.
`Application.Current!.Styles.Add(new Avalonia.Themes.Fluent.FluentTheme());`
